akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -326,6 +326,10 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
   var joinSeedNodesDeadline: Option[Deadline] = None
   var leaderActionCounter = 0
 
+  // to be able to accept a Welcome from previously attempted to join nodes
+  // even if we already started a new seedNodeProcess
+  var allowWelcomeFromJoiningSeedNodeProcesses: Set[Address] = Set.empty
```

Comment:
I don't think this is needed. If we receive a Welcome it was because we sent a Join previously. We assume cluster nodes are well behaved and don't do byzantine fault tolerance.

Reply:
Okey; and we're good to accept any Welcome that comes back then, sounds good to me if we want that.